5333: Endret visning av valg for å sende melding om saksbehandlingstid auto…

ivanskodje opened the pull request:
Ref: https://jira.adeo.no/browse/MELOSYS-5333

Endring er at vi bare vise denne delen når saken er knyttet til en bruker, sakstema er MEDLEMSKAP_LOVVALG, 
og behandlingstype er FØRSTEGANG.

Endringen er at "Melding om saksbehandlingstid" bare vises når de nye kriteriene er fylt.
![image](https://user-images.githubusercontent.com/6776092/192497693-20fea45f-f4b7-4501-b99d-8a8262079a04.png)

Mer spesifikt denne biten:
![image](https://user-images.githubusercontent.com/6776092/192497876-7911dd42-5833-4ba8-839e-aa9ee9c81196.png)

ivanskodje committed:
Endret visning av valg for å sende melding om saksbehandlingstid automatisk til nytt kodeverk.

ivanskodje committed:
Oppdatert tester og lagt til tester

ivanskodje committed:
Lagt til toggle!

janitamarielle commented on src/sider/journalforing/komponenter/journalforingform.js:
```diff
@@ -45,14 +45,25 @@ export const JournalforingForm = (props) => {
+    formValues.opprettnysak_behandlingstype &&
+    [MKV.Koder.behandlinger.behandlingstyper.FØRSTEGANG].includes(formValues.opprettnysak_behandlingstype);
+
+  if (!behandleAlleSakerToggleEnabled) {
```
Det var ikke så lett å se at visForvaltningsmelding ble satt annerledes hvis toggle var på. Gjør alt på ett sted?

ivanskodje replied:
Dette var gjort slik at vi når vi tar vekk toggle i fremtiden  slipper vi å måtte endre og være forsiktig med det man tar vekk. Vi sletter bare hele if og innholdet uten å røre noe annet.

ivanskodje replied:
Fiksa denne samtidig som jeg fiksa den andre kommentaren din smile
For å redusere koderot tok jeg ut logikken i en egen funksjon

janitamarielle commented on src/sider/journalforing/komponenter/journalforingform.js:
```diff
@@ -45,14 +45,25 @@ export const JournalforingForm = (props) => {
-    formValues.journalforingGjelder === BRUKER;
+    formValues.journalforingGjelder === BRUKER &&
+    formValues.sakstema &&
+    [MKV.Koder.sakstemaer.MEDLEMSKAP_LOVVALG].includes(formValues.sakstema) &&
```
Ikke bruk includes dersom det kun er en kode man skal sjekke opp mot

ivanskodje committed:
Tatt vekk include med bare én

ivanskodje committed:
Endret toggle samtidig som jeg gjorde andre review endringer.

Rettet på skrivefeil (forvaltningsmelding er ett ord).

Tatt logikken som avgjør om vi skal vise forvaltning i egen funksjon

janitamarielle commented on src/sider/journalforing/komponenter/journalforingform.js:
```diff
@@ -27,6 +27,29 @@ import "./journalforingform.css";
+      formValues.sakstema === MKV.Koder.sakstemaer.MEDLEMSKAP_LOVVALG &&
+      formValues.opprettnysak_behandlingstype &&
+      formValues.opprettnysak_behandlingstype === MKV.Koder.behandlinger.behandlingstyper.FØRSTEGANG
+    );
```
Slipper å sjekke om sakstema og opprettnysak_behandlingstype finnes. Dersom de er undefined eller null er de ikke lik MEDLEMSKAP_LOVVALG eller FØRSTEGANG

janitamarielle commented on src/sider/journalforing/komponenter/journalforingform.test.js:
```diff
@@ -49,30 +43,43 @@ describe("JournalforingForm", () => {
     };
   });
 
-  each([
```
Hvis du orker så kan du la disse stå bare med props.behandleAlleSakerToggleenabled = false så får vi testet at det funker med toggle av også

ivanskodje replied:
Jeg la til den gamle testen i en egen testfil, så vi ikke roter ny og gammel kode. Da blir det ekstra lett å slette den gamle  når vi tar vekk toggle :+1:

ivanskodje committed:
Tatt vekk unødvendig sjekk

ivanskodje committed:
Merge remote-tracking branch 'origin/master' into MELOSYS-5333

ivanskodje committed:
Lagt til den gamle testen som verifiserer tidligere oppførsel, samtidig som det er lett å ta den vekk
